cmd/crank/build.go

```diff
@@ -46,49 +47,62 @@ type buildCmd struct {
 }
 
 // Run runs the build cmd.
-func (c *buildCmd) Run(child *buildChild) error {
+func (c *buildCmd) Run(child *buildChild, logger logging.Logger) error {
+	logger = logger.WithValues("Name", child.name)
 	root, err := filepath.Abs(c.PackageRoot)
 	if err != nil {
 		return err
 	}
 
 	metaScheme, err := xpkg.BuildMetaScheme()
 	if err != nil {
+		logger.Info("Failed to build meta scheme for package parser", "error", err)
 		return errors.New("cannot build meta scheme for package parser")
 	}
+	logger.Info("Successfully built meta sceme for package parser")
 	objScheme, err := xpkg.BuildObjectScheme()
 	if err != nil {
 		return errors.New("cannot build object scheme for package parser")
 	}
+	logger.Info("Successfully built Object scheme for package parser")
 	img, err := xpkg.Build(context.Background(),
 		parser.NewFsBackend(child.fs, parser.FsDir(root), parser.FsFilters(buildFilters(root, c.Ignore)...)),
 		parser.New(metaScheme, objScheme),
 		child.linter)
 	if err != nil {
+		logger.Info(errBuildPackage, "error", err)
 		return errors.Wrap(err, errBuildPackage)
 	}
+	logger.Info("Successfully built package")
 
 	hash, err := img.Digest()
 	if err != nil {
+		logger.Info(errImageDigest, "error", err)
 		return errors.Wrap(err, errImageDigest)
 	}
-
+	logger.Info("Successfully found package digest")
 	pkgName := child.name
 	if pkgName == "" {
 		metaPath := filepath.Join(root, xpkg.MetaFile)
 		pkgName, err = xpkg.ParseNameFromMeta(child.fs, metaPath)
 		if err != nil {
+			logger.Info(errGetNameFromMeta, "error", err)
 			return errors.Wrap(err, errGetNameFromMeta)
 		}
 		pkgName = xpkg.FriendlyID(pkgName, hash.Hex)
 	}
 
 	f, err := child.fs.Create(xpkg.BuildPath(root, pkgName))
 	if err != nil {
+		logger.Info(errCreatePackage, "error", err)
 		return errors.Wrap(err, errCreatePackage)
 	}
+	logger.Info("Successfully created package image file")
 	defer func() { _ = f.Close() }()
-	return tarball.Write(nil, img, f)
+	if err := tarball.Write(nil, img, f); err != nil {
+		logger.Info("Failed to write package image", "error", err)
```

Comment:
I think we will want to return error here :)

Reply:
Done. PTAL.